### PR TITLE
Controller testing

### PR DESCRIPTION
**Pull request by Chike1990**
In this Milestone, I:
added RSpec as a gem to my project;
created a Request spec file for all my controllers;

wrote tests to check:
If the response status was correct;
If a correct template was rendered;
If the response body includes correct placeholder text.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+
```
Comment (SajjadAhmad14) [inline .rubocop.yml:1]: There are some tests failing. Kindly make sure that all the tests are passing.
fails

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Chibuike, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
PR has a title and summary. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: .rubocop.yml (hunk 1 above)
Comment (Chike1990) [reply .rubocop.yml:1]: @SajjadAhmad14, you would notice that failing tests are for the models which were last week's task. This pull request is for controller testing.
 I will fix that over the weekend. Looking at this now will inadvertently impede me from completing this week's tasks. Thank you

File: spec/requests/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "renders 'index' template" do
+      expect(response).to render_template(:index)
+    end
+
+    it 'should render the correct text in the index template' do
+      expect(response.body).to include('Here is the users/index<')
+    end
```
Comment (alexander16108) [inline spec/requests/users_controller_spec.rb]: you should ensure that the response body includes the correct placeholder text `expect(response.body).to include("Welcome to user index page")`

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @Chike1990 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- good pr ✔️ 
- passing linters ✔️ 
- required test passing ✔️ 


 ## Required Changes ♻️
_Check the comments under the review._
- I strongly recommend you fix those failing linters cause it may cause some trouble for you later on 👍🏾 💯 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @Chike1990, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**